src/platform/i386/boot_comp.c

```diff
@@ -19,6 +19,8 @@ extern u8_t *boot_comp_pgd;
 void *thd_mem[NUM_CPU], *tcap_mem[NUM_CPU];
 struct captbl *glb_boot_ct;
 
+thdid_t tid;
```

Comment:
Does this say that your changes to make thdid assignment at user-level only apply for x86-64? Does i386 compile? How does this impact ARM? My assumption was that it would be a direct change to all architectres, but I'm sure I missed some details.

Reply:
Any comment here to describe its background seems to be helpful.